container/container-core/src/main/java/org/talend/sdk/component/container/ContainerManager.java

```diff
@@ -111,13 +113,19 @@ public ContainerManager(final DependenciesResolutionConfiguration dependenciesRe
         this.logInfoLevelMapping = logInfoLevelMapping;
         this.containerInitializer = containerInitializer;
         this.resolver = dependenciesResolutionConfiguration.getResolver();
-        this.rootRepositoryLocation = ofNullable(dependenciesResolutionConfiguration.getRootRepositoryLocation())
-                .filter(Files::exists)
-                .orElseGet(() -> PathFactory.get(System.getProperty("user.home", "")).resolve(".m2/repository"));
 
-        if (log.isDebugEnabled()) {
-            log.debug("Using root repository: " + this.rootRepositoryLocation.toAbsolutePath());
+        Path rootRepo = ofNullable(dependenciesResolutionConfiguration.getRootRepositoryLocation())
+                .filter(Files::exists)
+                .orElseGet(() -> PathFactory.get(System.getProperty("user.home")).resolve(".m2/repository"));
+        // if we've defaulted to user home m2 (fallback), we want to check if we're in running in a fatjar
+        if (PathFactory.get(System.getProperty("user.home")).resolve(".m2/repository").equals(rootRepo)) {
```

Comment:
M2 repository path can be ovewritten by <localRepository> in settings.xml file
=> ${user.home}/.m2/repository might not be a path to the repository.